pennylane/_device.py

```diff
@@ -351,7 +345,6 @@ def apply(self, operation, wires, par):
             wires (Sequence[int]): subsystems the operation is applied on
             par (tuple): parameters for the operation
         """
-        raise NotImplementedError
```

Comment:
I think this one should remain, since `check_validity` is not abstract.

Reply:
Thank you for spotting that!